### ISSUE TITLE
Whisper V3 PEFT finetuning is broken

### ISSUE THREAD

**Issue by nshmyrev** [huggingface/peft#1223]
### System Info

transformers          4.36.0.dev0, 
peft                          0.6.3.dev0
accelerate                    0.25.0
bitsandbytes                  0.41.2.post2

### Who can help?

@sanchit-gandhi 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

After finetuning with LORA following this guide just replacing v2 with v3 everywhere, the model produces garbage

https://github.com/huggingface/peft/blob/main/examples/int8_training/peft_bnb_whisper_large_v2_training.ipynb

the suspicious output line is

```Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.```

### Expected behavior

V3 model finetunes as V2

**Comment by nshmyrev** [comment]
It seems it just diverged, with 16-bit its ok.

**Comment by akuzeee** [comment]
Hello @nshmyrev,

I am facing the same issue. Could you please let me know which parts of the notebook you changed?

In particular, it appears that 16-bit is already being used during training (as fp16=True is set within Seq2SeqTrainingArguments). Are you referring to switching to 16-bit during inference?

**Comment by sanchit-gandhi** [comment]
I would strongly recommend using fp16 for inference: https://github.com/huggingface/peft/discussions/477#discussion-5213394

**Comment by Po-Kai** [comment]
I got the same problem, when I change whisper-large-v2 to v3, It's broken, even I inference on fp16 or fp32.

**Comment by nshmyrev** [comment]
For 16 bit i mean 16-bit lora. Yes, it requires gradient accumulation during training and batch size is smaller, but it is more stable.
